autoload/ale/util.vim

```diff
@@ -571,3 +571,31 @@ function! ale#util#ToResource(uri) abort
 
     return l:resource
 endfunction
+
+" Returns a list of matches of pattern in string.
+function! ale#util#MatchStrings(string, pattern)
```

Comment:
Let's keep the completion-specific code in the completion script so it isn't loaded for people who don't use completion.

Reply:
Yes that would be more efficient. And if the functions `MatchStrings` and `GetCWordBoundary` can be reused for other purposes, we could always put them in `util.vim`.